kippenhahn/kippenhahn.py

```diff
@@ -968,16 +990,17 @@ def plot(self, ax=None):
 
 
 
-        if self._param['signed_log_cmap'] and (self._cmap_label is not None):
+        if self._param['signed_log_cmap'] and (cmap_label is not None):
             cmap_label = "sign x log(max(1,abs(" + cmap_label[4:]+"))"
 
 
-        if ax is None:
+        if self._param['ax'] is None:
             fig1 = plt.figure()
             ax1 = fig1.add_subplot(111)
             fig1.subplots_adjust(top=0.80, left=0.12, right=0.9, bottom=0.12)
         else:
-            ax1 = ax
+            ax1 = self._param['ax'] 
+            fig1 = self._param['fig'] 
```

Comment:
After this change, shouldn't we remove the argument `ax` from the plot function? If it is still used somewhere else you should replace it there with the new axis object stored in the kippenhahn object.